packages/client/src/generation/generateClient.ts

```diff
@@ -301,18 +301,21 @@ export async function generateClient({
     await writeFile(packageJsonTargetPath, pkgJson)
   }
 
-  if (!testMode && process.env.INIT_CWD) {
-    const backupPath = path.join(process.env.INIT_CWD, 'node_modules/.prisma/client')
-    if (finalOutputDir !== backupPath && !generator?.isCustomOutput) {
-      await copy({
-        from: finalOutputDir,
-        to: backupPath,
-        recursive: true,
-        parallelJobs: process.platform === 'win32' ? 1 : 20,
-        overwrite: true,
-      })
-    }
-  }
+  // if (!testMode && process.env.INIT_CWD) {
```

Comment:
This has been reported by redwood as being incorrect. It causes the generator to copy the generated files to `backupPath`. As far as I understand, `backupPath` will cause to the copy to happen wherever you run `prisma generate`. I don't see how this could be useful since the we already generated the client where it should be in `finalOutputDir`.

(help needed, cc @Jolg42 @aqrln 🙏)

Reply:
It looks like this code was added to support pnpm, see commit https://github.com/prisma/prisma/commit/2be6b71dfa27202058eeff5274dbaf0aabc31c73

Reply:
Thanks, I don't think this is actually needed for `pnpm`. What this does is to copy the contents from the actual final directory to the root of `node_modules`. That cannot be used by `pnpm` and is causing pollution on yarn (and probably others).

So before this PR, you would get an unnecessary generated artifact copied in the root of `node_modules`.
![Screenshot from 2022-03-21 11-16-17](https://user-images.githubusercontent.com/18401805/159242180-0b775a93-502e-4a8e-87d6-be47735a2d01.png)

Similarly, this is causing worse directory pollution by creating a `node_modules` folder and copying the `.prisma/client` in it, as described in the original issue https://github.com/prisma/prisma/issues/12083

And after this PR, you don't get a polluted `pnpm` `node_modules` or `yarn` workspace.
![Screenshot from 2022-03-21 11-17-17](https://user-images.githubusercontent.com/18401805/159242708-7b530ec0-b743-4c67-aeb1-9cdcbd2ca40d.png)

The client will get generated where it belongs, and only wherever it has been hoisted.
![Screenshot from 2022-03-21 11-16-17](https://user-images.githubusercontent.com/18401805/159243031-0b09d50e-81e3-4796-8dfb-d42aef6a11ff.png)

Happy to set a call up today if needed, I need to merge this soon for a patch :)

Reply:
All E2E tests are passing except for one with `pnpm` because I was explicitly checking for the existence of `node_modules/.prisma/client` when I should have check that to be in the hoisted folder(like in the screenshot just above)  :facepalm:

Reply:
Nice! Then it sounds like we can get rid of this 🧹

Reply:
Note that there is some related code in `getDotPrismaDir` about `INIT_CWD` and pnpm https://github.com/prisma/prisma/blob/integration/fix/yarn1-workspaces/packages/client/src/generation/generateClient.ts#L111-L126

Reply:
Added it as a small item to our deep dive sessions :diving_mask: